modules/Rock.cmake

```diff
@@ -1350,6 +1368,39 @@ function(rock_gtest TARGET_NAME)
 
     rock_setup_gtest_test(${TARGET_NAME} ${GMOCK_DIR} ${GTEST_DIR})
     rock_add_test(${TARGET_NAME} "${__rock_test_parameters}")
+    if (ROCK_TEST_CXX_GCOVR_GENERATION_ENABLED)
+        rock_coverage_report_generation(${PROJECT_NAME} ${TEST_TARGET_NAME})
+    endif()
+endfunction()
+
+function(rock_coverage_report_generation TARGET_NAME TEST_TARGET)
+    message(STATUS "Generating report for ${TARGET_NAME} package")
+    find_program(
+        gcovr_exec NAMES
+        ${ROCK_GCOVR_EXECUTABLE} gcovr
+    )
+    if (NOT gcovr_exec)
+        message(FATAL_ERROR "Could not find an executable for gcovr.")
+    endif()
+
+    set(gcovr_config_option --filter ${PROJECT_SOURCE_DIR}/src/ --html -o ${PROJECT_SOURCE_DIR}/coverage.html)
+    if(${ROCK_COVERAGE_REPORT_PATH})
+        message(warning "Setting an explicit config path coverage report")
+        set(gcovr_config_option --filter ${PROJECT_SOURCE_DIR}/src/ --html -o ${ROCK_COVERAGE_REPORT_PATH}/coverage.html)
+    endif()
+
+    if(ROCK_TEST_CXX_SONARQUBE_COVERAGE_ENABLED AND ROCK_TEST_CXX_GCOVR_GENERATION_ENABLED)
```

Comment:
I think that having options specific to SonarQube here is pushing it a bit. It would be better to have a variable that allows passing arbitrary options to gcovr, and have our autoproj configuration set them for sonarqube.